### ISSUE TITLE
Allow locus-tags that are longer than 12 alphanumeric characters, contain some special characters

### ISSUE THREAD

**Issue by taylorreiter** [oschwengers/bakta#92]
Hello! Thank you again for all of the work you've put into bakta. I'm writing to ask if it would be possible to allow the `--locus-tag` parameter to allow more than 12 alphanumeric characters.

When I annotate publicly available genomes, I like to use there accession identifier as the locus tag:

```
bakta --db inputs/bakta_db/db/ --prefix GCF_014267685.1 --output outputs/gtdb_genomes_bakta/s__Bacillus_pumilus  \
    --locus GCF_014267685.1 --locus-tag GCF_014267685.1 --keep-contig-headers \
    outputs/gtdb_genomes_charcoa;l/s__Bacillus_pumilus/GCF_014267685.1_genomic.fna.gz.clean.fa.gz
```

However, this currently produces the following error:
```
ERROR: invalid 'locus-tag' parameter (GCF_007674165.1)!
Locus tag prefixes must contain between 3 and 12 alphanumeric uppercase characters and start with a letter.
```

Would it be possible to allow a locus tag like this? 

One of my downstream use cases is feeding these genomes into roary, so retaining compatibility with that is also desirable

**Comment by oschwengers** [comment]
Hello @taylorreiter,
sure - that's not much of an issue. Sticking to the strict INSDC format as a default has been more of a design decision and I also stumbled over this a couple of times in my own analyses. So I totally see the point. I will relax the `locus_tag` prefix tests to a more flexible format.

Would alphanumerics and `_.-` (`[a-zA-Z0-9_.-]{1,24}`) be OK? Are there any characters that should also be allowed?

This would be the default and in INSDC compliant mode (`--compliant`), the strict INSDC format would be used.

**Comment by oschwengers** [comment]
I also tested Roary `v3.13.0` using Bakta's `*.gff3` output annotating `GCF_014267685.1` and some additional `Bacillus pumilus` genomes w/o any issues.

**Comment by taylorreiter** [comment]
Thank you so much @oschwengers,this is wonderful! `[a-zA-Z0-9_.-]{1,24}` would be perfect, I can't think of any accessions that don't fall within that pattern.

**Comment by oschwengers** [comment]
A fix is now available in `main`. It might take a while for the next release but meanwhile you could install & try it within a conda environment via:

```bash
git clone https://github.com/oschwengers/bakta.git
python -m pip install --no-deps --ignore-installed bakta/
```

Please let me know if this is working for you.

**Comment by oschwengers** [comment]
@taylorreiter FYI: the relaxed `locus-tag` checks are now available in [v1.3.1](https://github.com/oschwengers/bakta/releases/tag/v1.3.1).

It should be available in a few hours so I'll close this for now. However, please do not hesitate to re-open it in any case. Best regards!

**Comment by taylorreiter** [comment]
Just got a chance to test it from the conda install, and it works perfectly! Thank you so much!
